auth-tokens/src/main/java/com/palantir/tokens/auth/UuidStringConverter.java

```diff
@@ -27,20 +27,19 @@
  *
  * https://github.com/FasterXML/jackson-databind/blob/master/src/main/resources/META-INF/LICENSE
  */
-class UuidStringConverter {
-    private final static char[] HEX_CHARS = "0123456789abcdef".toCharArray();
+final class UuidStringConverter {
+    private static final char[] HEX_CHARS = "0123456789abcdef".toCharArray();
 
     private UuidStringConverter() {}
 
     static String toString(UUID value) {
         final char[] ch = new char[36];
         final long msb = value.getMostSignificantBits();
-        writeInt((int) (msb >> 32), ch, 0);
+        writeInt((int) (msb >>> 32), ch, 0);
         ch[8] = '-';
-        int i = (int) msb;
-        writeShort(i >>> 16, ch, 9);
+        writeShort((int) (msb >>> 16), ch, 9);
```

Comment:
this is definitely not equivalent to what was there before

Reply:
explicitly, (int) (-2^32 >>> 16) = 0, but ((int) -2^32)  >>> 16 = 2^16

Reply:
please can we just leave this logic the way it was? this is directly copied from Jackson and given that I don't have a decent test suite I'm not convinced of safety

Reply:
so they're not equivalent ints but they are equivalent shorts, which is what we care about here. fine going back to the original logic and just giving the variable a real name though so we keep this as close to the original as possible.